server/src/recipe/index.js

```diff
@@ -1,13 +1,21 @@
-import { createRecipe, deleteRecipe, listPublishedRecipes, listUserRecipes } from './recipe.controller';
-
+import {
+  createRecipe,
+  deleteRecipe,
+  listPublishedRecipes,
+  listUserRecipes,
+  publishRecipe,
+  unpublishRecipe,
+} from './recipe.controller';
 import { Router } from 'express';
 
 const router = Router();
 const path = '/recipes';
 router
-  .post('/delete', deleteRecipe)
   .get('/published', listPublishedRecipes)
   .get('/userRecipes', listUserRecipes)
-  .post('/', createRecipe);
+  .post('/', createRecipe)
+  .post('/delete', deleteRecipe)
+  .post('/publish', publishRecipe)
+  .post('/unpublish', unpublishRecipe);
```

Comment:
you can just create one post method and call it `updateIsPublished` with parameters being recipeId and boolean value called isPublished and you just update the recipe with recipeId to have isPublished be the value of isPublished